### ISSUE TITLE
[clang] Segfault with `-module-file-info`

### ISSUE THREAD

**Issue by chestnykh** [llvm/llvm-project#98365]
If clang is invoked with `-module-file-info` and the file is of incorrect type then clang segfaults.

There is assert in the code
`assert(isCurrentFileAST() && "dumping non-AST?");`
Should this behavior be fixed or left unchanged?

CC: @AaronBallman @cor3ntin @Endilll

**Comment by shafik** [comment]
CC @ChuanqiXu9

**Comment by ChuanqiXu9** [comment]
Yeah, it will be better to use a diagnostic than crashing here. The reason why it is left so long is that I thought the tool is majorly used by developers not end users. But we provide it after all. And I feel this issue is suit for beginners since beyond the background, it simply requires us to add a diagnostic for it.

**Comment by chestnykh** [comment]
> Yeah, it will be better to use a diagnostic than crashing here. The reason why it is left so long is that I thought the tool is majorly used by developers not end users. But we provide it after all. And I feel this issue is suit for beginners since beyond the background, it simply requires us to add a diagnostic for it.

Okay, I will work on this issue. Assign please it to me
